lesson28/src/js/forgot-password.js

```diff
@@ -0,0 +1,56 @@
+import * as validation from "./validation";
+
+const submitButton  = document.querySelector('[data-id="submit_button"]');
+const email = document.querySelector('[data-id="email"]');
+
+submitButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    checkRegisteredEmail(email);
+});
+
+const changeDisabledStatusSubmitButton = () => submitButton.disabled = validation.isInvalid();
+
+const validateEmail = (input) => {
+    input.classList.add("field-invalid");
+
+    input.addEventListener("blur", () => {
+        if(input.hasAttribute("required") && input.value.trim() === ""){
+            validation.showErrorMessage(input,"入力してください");
+        }else{
+            validation.removeErrorMessage(input);
+            validation.checkEmail(input);
+        }
+
+        changeDisabledStatusSubmitButton();
+    });
+
+    input.addEventListener("input" , () => {
+        if(input.nextElementSibling && input.nextElementSibling.classList.contains("field-invalid")){
+            validation.checkEmail(input);
+        }
+    
+        if(input.hasAttribute("required") && input.value.trim() === ""){
+            validation.removeErrorMessage(input);
+            validation.showErrorMessage(input,"入力してください");
+        }
+
+        changeDisabledStatusSubmitButton();
+    });
+}
+
+validateEmail(email);
```

Comment:
The implementation of the validation is much the same as on the other pages. 
We have used `input` as an argument so that the input element can be determined at the time of the call, but we think that the argument may not be necessary.